pandas/core/dtypes/inference.py

```diff
@@ -263,7 +263,8 @@ def is_list_like(obj):
     """
 
     return (hasattr(obj, '__iter__') and
-            not isinstance(obj, string_and_binary_types))
+            not isinstance(obj, string_and_binary_types) and
+            isinstance(obj, collections.Iterable))
 
 
```

Comment:
pls run a sub-set of the perf tests (IOW something like this)

```
asv continuous master thisbranchname --bench indexing
asv continuous master thisbranchname --bench timeseries
```
and report any significant diffs.

This touches code from almost everywhere indirectly so want to make sure no regressions.